tests/compile-fail/while_loop.rs

```diff
@@ -53,6 +53,23 @@ fn main() {
     while let Some(x) = y { // no error, obviously
         println!("{}", x);
     }
+
+
+    while let Option::Some(x) = (1..20).next() { //~ERROR this loop could be written as a `for` loop
```

Comment:
Since these examples depend on a temporary, they are all actually infinite loops :smile: 

I'd prefer if we used vector iteration here or stored the ranges in an external mutable variable.